pkg/volume/qingcloud_volume/attacher.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package qingcloud_volume
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/qingcloud"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+)
+
+type qingcloudVolumeAttacher struct {
+	host      volume.VolumeHost
+	qcVolumes qingcloud.Volumes
+}
+
+var _ volume.Attacher = &qingcloudVolumeAttacher{}
+
+var _ volume.AttachableVolumePlugin = &qingcloudVolumePlugin{}
+
+func (plugin *qingcloudVolumePlugin) NewAttacher() (volume.Attacher, error) {
```

Comment:
? NewAttacher is just a method for volume.AttachableVolumePlugin, I refer to aws